### PR TITLE
✨ Add an artifact loader for `.yaml`

### PR DESCRIPTION
**Pull request by rcannood**
* Fix incorrect documentation for `load_image`
* Add loaders for `.gif` and `.yaml`.

In addition, we could add a data loader for `.rds`, which when called, produces an error message that you should use laminr for loading `.rds` files. WDYT?

### CODE REVIEW

**Comment by rcannood** [comment]
FYI -- laminlabs/laminr#81 added a data loader for .rds.

**Comment by falexwolf** [comment]
Thanks, Robrecht! Great!

> In addition, we could add a data loader for .rds, which when called, produces an error message that you should use laminr for loading .rds files. WDYT?

That's a nice idea, yeah!

**Comment by falexwolf** [comment]
We need a test here for the new `.yaml` artifact loader:

https://github.com/laminlabs/lamindb/blob/996a41640d693cd5254170a65aa140ac5078f416/tests/core/test_artifact.py#L714

@Koncopd, do you want to finalize and rename the test function to `test_artifact_loaders()`?

**Comment by Koncopd** [comment]
Do we want to add `pyyaml` to the deps?

**Comment by falexwolf** [comment]
No, I'd not add pyyaml to the dependencies because we have too many anyway and nothing in lamindb requires it.

**Comment by Koncopd** [comment]
Ok, i wanted to push some things here but again we have the problem with CI for PRs not from this repo.

**Comment by Koncopd** [comment]
> Thanks, Robrecht! Great!
> 
> > In addition, we could add a data loader for .rds, which when called, produces an error message that you should use laminr for loading .rds files. WDYT?
> 
> That's a nice idea, yeah!

@rcannood @falexwolf i think a warning is more reasonable for that https://github.com/laminlabs/lamindb/pull/2188

File: lamindb/core/loaders.py [hunk 1]
```diff
@@ -107,9 +108,17 @@ def load_json(path: UPathStr) -> dict:
         data = json.load(f)
     return data
 
+def load_yaml(path: UPathStr) -> dict:
+    """Load `.yaml` to `dict`."""
+    import yaml
```
Comment (Koncopd) [inline lamindb/core/loaders.py:113]: Maybe except `ImportError` and log a warning that the package is not installed and return a path.

**Comment by Koncopd** [comment]
`load_yaml` needs a test. 

There is also a comment from me, otherwise good to merge when the comments are resolved.

**Comment by Koncopd** [comment]
This looks inactive, so moving it here https://github.com/laminlabs/lamindb/pull/2270

**Pull request closed without merging**